File: Wox.UI.React/src/enums/WoxLastQueryModeEnum.ts
import { BaseEnum } from "./base/BaseEnum.ts"

export type WoxLastQueryMode = string

export class WoxLastQueryModeEnum extends BaseEnum {
  static readonly WoxLastQueryModePreserve = WoxLastQueryModeEnum.define("preserve", "preserve")
  static readonly WoxLastQueryModeEmpty = WoxLastQueryModeEnum.define("empty", "empty")
}

File: wox.ui.flutter/wox/lib/enums/wox_last_query_mode_enum.dart
typedef WoxLastQueryMode = String;

enum WoxLastQueryModeEnum {
  WOX_LAST_QUERY_MODE_PRESERVE("preserve", "preserve"),
  WOX_LAST_QUERY_MODE_EMPTY("empty", "empty");

  final String code;
  final String value;

  const WoxLastQueryModeEnum(this.code, this.value);

  static String getValue(String code) => WoxLastQueryModeEnum.values.firstWhere((activity) => activity.code == code).value;
}
